make/modules/java.base/Gendata.gmk

```diff
@@ -71,6 +71,9 @@ $(GENDATA_CACERTS): $(BUILD_TOOLS_JDK) $(wildcard $(GENDATA_CACERTS_SRC)/*)
 ifeq ($(CACERTS_FILE), )
   TARGETS += $(GENDATA_CACERTS)
 endif
+ifneq ($(CACERTS_SRC), )
+  GENDATA_CACERTS_SRC := $(CACERTS_SRC)
+endif
```

Comment:
Does this even work?! You are reassigning the variable after it has been used. The := assignment means that it not a macro.

Reply:
I would have expected to see something like:
```
ifneq ($(CACERTS_SRC), )
  GENDATA_CACERTS_SRC := $(CACERTS_SRC)
else
  GENDATA_CACERTS_SRC := $(TOPDIR)/make/data/cacerts/
endif
```
at line 63.

Reply:
my assumption was the recipe gets resolved later

Reply:
this was my understanding: https://www.gnu.org/software/make/manual/html_node/Variables-in-Recipes.html
```
This occurs after make has finished reading all the makefiles and the target is determined to be out of date; so, the recipes for targets which are not rebuilt are never expanded. 
```
but i'm going to double check I was checking the resultant cacerts correctly in my tests

Reply:
Oh, I didn't expand the diff far enough to actually see the context correctly when I reviewed this as I would never have imagined the conditional to be placed after the rule. While this will work as so far as using the correct files, incremental builds will not be correct, because the rules are defined in the first pass.

I very much agree with Magnus that this conditional belongs around line 63.

Reply:
yes, thanks, feeling rather stupid here! i'll raise an issue to fix

Reply:
@andrew-m-leonard Don't be. Make is a horrible programming language, both syntactically and semantically. It's taken me years to be somewhat comfortable with it, and often I just manage to get it right only by sticking to a few, well-proven and battle-hardened patterns. :)